core/domain/exp_services.py

```diff
@@ -1009,11 +1020,12 @@ def _save_exploration(
         Exception. The versions of the given exploration and the currently
             stored exploration model do not match.
     """
-    exploration_rights = rights_manager.get_exploration_rights(exploration.id)  # type: ignore[no-untyped-call]
-    if exploration_rights.status != rights_domain.ACTIVITY_STATUS_PRIVATE:
-        exploration.validate(strict=True)  # type: ignore[no-untyped-call]
-    else:
-        exploration.validate()  # type: ignore[no-untyped-call]
+    exploration_rights = rights_manager.get_exploration_rights(exploration.id)
+    exploration.validate(
+        strict=(
+            exploration_rights.status != rights_domain.ACTIVITY_STATUS_PRIVATE
```

Comment:
Could you pull this out into a new local variable instead, defined before this line? What you've written here is a bit complex and makes an important part of the code harder to understand.

Reply:
Done, the condition is pulled out into a new variable and passed that variable for strict.